API_changes.rst

```diff
@@ -6,6 +6,7 @@ PyModbus - API changes.
 Version 3.4.2
 -------------
 -ModbusSerialServer now accepts request_tracer=.
+-ModbusSerialServer does not use loop parameter anymore
```

Comment:
I think you have changed a lot more parameters, every API change needs to be documented.

Please add a new headline 
Version 3.5.0 (future)

since we do not accept API changes in 3.4.x versions.

Reply:
Thank you for the last review. I have added all API changes as requested.